src/components/FilmsList/FilmsList.js

```diff
@@ -1,18 +1,42 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+
+import { store } from '../../store/index';
 import './FilmsList.scss';
 import { FilmCard } from '../FilmCard';
 
-export const FilmsList = (props) => {
-  const { films } = props;
+export class FilmsList extends React.Component {
+  state = {
+    films: [],
+  };
+
+  unsubscribe = null;
+
+  componentDidMount() {
+    this.unsubscribe = store.subscribe(this.getFilmsList);
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  getFilmsList = () => {
+    this.setState({
+      films: [...store.getState().films],
+    });
+  }
+
+  render() {
+    const { films } = this.state;
 
-  return (
-    <div className="films">
-      {films.map(film => (
-        <FilmCard key={film.id} {...film} />
-      ))}
-    </div>
-  );
+    return (
+      <div className="films">
+        {films.map(film => (
+          <FilmCard key={film.id} {...film} />
```

Comment:
if you have the same movies, you'll have the same id, which makes key the same too

Reply:
I added lodash library